### ISSUE TITLE
Feature Request: "Save unsubmitted form data before suspending"

### ISSUE THREAD

**Issue by oucil** [Hau-Hau/firefox-tab-suspender#53]
I often have to leave forms on one tab to continue to research on others, but when I get back the tab has been suspended and when reloaded is empty, it's very frustrating because other than pinning the tab which is not an adequate solution, there is currently no way to avoid it other than shutting off the add-on.  Please have the suspend process save a copy of any un-submitted form data and repopulate the form once it's reloaded.

Thanks!

**Comment by Adam-0-Moore** [comment]
This please!

**Comment by Hau-Hau** [comment]
Hello!

Suspending tabs with unfinished forms also drives me crazy.
I have this feature in plans, but first I need fix some critical bugs.

**Comment by atomicwrites** [comment]
I think a stopgap solution might be to not suspend tabs with edited form fields, until something better can be done.

**Comment by leandrotapia** [comment]
Hi!

I think it could be achieved by having a "Keep alive" flag option to mark the tab as "Do not suspend" on tab context menu.
After that, there are similar cases that can be solved in a similar way as I mentioned before. For example, you'd be reading a PDF on Google Drive, or editing a Google Doc, and you don't want the tab to be suspended because you'd lose the document position when the tab was restored.

So It could not be a "form" issue rather than a working tab that would not have to be suspended.

I'm a person who most the time has a lot of opened tabs, so I understand it.
In the other hand, I started to think about having a "working place" and I found an add-on for grouping tabs, so I made a group for a "working in" tabs, and other tabs (those I'm not working in at that moment) are in different groups which can be suspended.

Then it could be reduced to a "Keep alive those tabs I'm working in" problem.

Regards!.

**Comment by oucil** [comment]
@leandrotapia I submitted the Keep Alive feature request (#52) separately because they are not the same issue.  Just because I have a form going doesn't mean I don't want it suspended, I just want a form data snapshot first.

With regard to tab groups, again, not related to this feature request, that's a totally separate add-on that people may or may not have, and has nothing to do with form data.

**Comment by blueyed** [comment]
Yeah, it should not suspend tabs with changed/pending form data by default.
Just for reference: https://github.com/rNeomy/auto-tab-discard supports this.
